container/container_unix.go

```diff
@@ -345,7 +346,10 @@ func (container *Container) UnmountVolumes(forceSyscall bool, volumeEventLog fun
 
 		if volumeMount.Volume != nil {
 			if err := volumeMount.Volume.Unmount(volumeMount.ID); err != nil {
-				return err
+				logrus.Errorf("%s Unmount failed %v", volumeMount.Volume.Name(), err)
+				// on error keep unmounting the rest of the volumes. Fixed in docker 1.13
+				unmountError = err
+				continue
 			}
 			volumeMount.ID = ""
```

Comment:
good question, the documentation is vague and mysterious. Seems to be a generated noncrypto ID that is set during the mount as part of the mount request. part of the MountPoint struct in volumes/volumes.go
```
	// ID is the opaque ID used to pass to the volume driver.
	// This should be set by calls to `Mount` and unset by calls to `Unmount`
	ID string
```